pandas/tests/reshape/test_cut.py

```diff
@@ -112,6 +112,14 @@ def test_bins_not_monotonic():
         cut(data, [0.1, 1.5, 1, 10])
 
 
+def test_bins_monotic_not_overflowing():
```

Comment:
Could you parameterize this test with timedelta and numeric data as well?

Reply:
1) I added the parametrization with `int64` type, should I add more integer types (`int32`, `int16`, etc.) ? The 64 one should cover the smaller precisions IMO.

2) Regarding floating types, there was no overflowing error in the first place, just a numpy Runtime warning but `cut` deals with `inf` bounds just fine so I did not add a test.
```
pd.cut([-1,1], bins=np.array([-65500, 0, 65500], dtype=np.float16))
numpy\core\fromnumeric.py:56: RuntimeWarning: overflow encountered in multiply
  return getattr(obj, method)(*args, **kwds)

[(-inf, 0.0], (0.0, inf]]
Categories (2, interval[float64]): [(-inf, 0.0] < (0.0, inf]]
```
(pandas `0.24.2`)

3) For timedelta data, are we talking about standard `datetime.timedelta`,  `pd.Timedelta`, `np.timedelta64`,  ? I added a test for the latter and it seems that `cut` cannot be used at the moment with the two remaining.

Reply:
`in64` is good enough imo. The parameterization looks good for timedelta.